.github/workflows/packer-build-and-publish.yml

```diff
@@ -147,14 +152,68 @@ jobs:
           path: ${{ steps.build.outputs.readme }}
           overwrite: true
 
+  copy:
+    runs-on: ubuntu-latest
+    needs:
+      - check-if-main
+      - build
+    env:
+      SOURCE_ACCOUNT: "654654387067" # dev
+      DEST_ACCOUNT: "654654387067" # dev
+    steps:
+      - name: Authenticate to ${{ env.SOURCE_ACCOUNT }}
+        uses: grafana/shared-workflows/actions/aws-auth@main
+        with:
+          role-arn: "arn:aws:iam::${{ env.SOURCE_ACCOUNT }}:role/github-actions/packer-role"
+          pass-claims: "repository_owner, repository_name, job_workflow_ref, workflow_ref, event_name"
+          set-creds-in-environment: true
+
+      - name: Share AMIs with ${{ env.DEST_ACCOUNT }}
+        run: |
+          IFS=',' read -ra REGION_AMI_PAIRS <<< "${{ needs.build.outputs.amis }}"
+          for PAIR in "${REGION_AMI_PAIRS[@]}"; do
+            AMI_ID="${PAIR##*:}"
+            echo "Sharing AMI $AMI_ID with account ${{ env.DEST_ACCOUNT }}"
+            aws ec2 modify-image-attribute \
+              --image-id $AMI_ID \
+              --launch-permission "Add=[{UserId=${{ env.DEST_ACCOUNT }}}]"
+          done
+
+      - name: Authenticate to ${{ env.DEST_ACCOUNT }}
+        uses: grafana/shared-workflows/actions/aws-auth@main
+        with:
+          role-arn: "arn:aws:iam::${{ env.DEST_ACCOUNT }}:role/github-actions/packer-role"
+          pass-claims: "repository_owner, repository_name, job_workflow_ref, workflow_ref, event_name"
+          set-creds-in-environment: true
+
+      - name: Copy Shared AMI to Destination Account
+        run: |
+          IFS=',' read -ra REGION_AMI_PAIRS <<< "${{ needs.build.outputs.amis }}"
+          for PAIR in "${REGION_AMI_PAIRS[@]}"; do
+            AMI_ID="${PAIR##*:}"
+            AMI_REGION="${PAIR%%:*}"
+            echo "Copying $AMI_ID to $AMI_REGION on ${{ env.DEST_ACCOUNT }}"
+            NEW_AMI_ID=$(aws ec2 copy-image \
```

Comment:
Why assign this to a variable if you don't do anything with it? Suggest printing it out.

Reply:
This is gone.